packages/cursorless-engine/src/util/getPartialTargetDescriptors.ts

```diff
@@ -23,11 +23,20 @@ export function getPartialTargetDescriptors(
     case "replace":
     case "editNew":
       return getPartialTargetDescriptorsFromDestination(action.destination);
+    case "parsed":
+      // FIXME: This is a hack
+      return action.arguments.filter(isPartialTargetDescriptor);
     default:
       return [action.target];
   }
 }
 
+function isPartialTargetDescriptor(
+  arg: unknown,
+): arg is PartialTargetDescriptor {
+  return typeof arg === "object" && arg != null && "type" in arg;
+}
+
```

Comment:
this is pretty brutal. I do kind of feel like targets are a bit special, and our system kinda breaks down when we have args and targets and destinations all mixed up

we could try the idea we discussed where we inject expected type when we parse placeholders, but that is also kind of rough

Idk. I'd be tempted to ship in either private or semi-private and sit on it

Reply:
If you're not happy with the solution then definitely go private